charms/worker/k8s/charmcraft.yaml

```diff
@@ -145,7 +145,11 @@ actions:
 
 parts:
   charm:
-    build-packages: [git]
+    plugin: charm
+    build-packages:
+    - git
+    - rustc
+    - cargo
```

Comment:
yup -- we need `rustc` and `cargo` to build libraries for pydantic... we need to confirm the charms built by focal work on jammy + noble

Reply:
Well, this firmly breaks jammy and noble installs because pydantic 2 gets compiled into a binary which can only be used by python38.

```
root@juju-fc684e-0:/var/lib/juju/agents/unit-k8s-0/charm/venv/pydantic_core# file _pydantic_core.cpython-38-x86_64-linux-gnu.so
_pydantic_core.cpython-38-x86_64-linux-gnu.so: ELF 64-bit LSB shared object, x86-64, version 1 (SYSV), dynamically linked, BuildID[sha1]=c09ad7f4f2d18c122a214f4c169330304771d064, stripped
```